src/events/pg.js

```diff
@@ -0,0 +1,122 @@
+ * @param {Function} wrappedFunction The pg's module
+ * @returns {Function} The wrapped function
+ */
+function pgClientWrapper(wrappedFunction) {
+    return function internalpgClientWrapper(queryString, arg1, arg2) {
+        // eslint-disable-next-line no-var
+        var patchedCallback;
+
+        const callback = (arg2 === undefined) ? arg1 : arg2;
+        const params = (arg1 instanceof Function) ? [] : arg1;
+        let sqlObj = {};
+
+        try {
+            sqlObj = sqlParser.parse(queryString);
+        } catch (error) {
+            tracer.addException(error);
+        }
+
+        let databaseType = 'local';
+
+        const {
+            type,
+            table,
+        } = sqlObj;
+
+        const {
+            database,
+            host,
+        } = this.connectionParameters;
+
+        if (host.match('.rds.')) { databaseType = 'rds'; }
+        if (host.match('.redshift.')) { databaseType = 'redshift'; }
+
+        try {
+            const resource = new serverlessEvent.Resource([
+                database, // name of the database
+                databaseType,
+                queryString,
+            ]);
+
+            const startTime = Date.now();
+
+            const dbapiEvent = new serverlessEvent.Event([
+                `dbapi-${uuid4()}`,
+                utils.createTimestampFromTime(startTime),
+                null,
+                'pg',
+                0,
+                errorCode.ErrorCode.OK,
+            ]);
+
+            dbapiEvent.setResource(resource);
+
+            const extendedData = {};
+            if ((type === 'select') || (!tracer.metadata_only)) {
+                extendedData.Query = queryString;
+            }
+
+            eventInterface.addToMetadata(dbapiEvent, {
+                Host: host,
+                Driver: 'pg',
+                Type: type,
+                'Table Name': table,
+            }, extendedData);
+
+
+            const responsePromise = new Promise((resolve) => {
+                patchedCallback = (err, res) => {
+                    if (err) {
+                        eventInterface.setException(dbapiEvent, err);
+                    } else {
+                        eventInterface.addToMetadata(dbapiEvent, {
+                            rowCount: res.rowCount,
+                        });
+                    }
+
+                    if (callback) {
+                        callback(err, res);
+                    }
+
+                    dbapiEvent.setDuration(utils.createDurationTimestamp(startTime));
+                    resolve();
+                };
+            }).catch((err) => {
+                tracer.addException(err);
+            });
+
+            tracer.addEvent(dbapiEvent, responsePromise);
+        } catch (error) {
+            tracer.addException(error);
+        }
+
+        try {
+            return wrappedFunction.apply(this, [queryString, params, patchedCallback]);
+        } catch (error) {
+            throw error;
```

Comment:
I don't think this can throw an error (it was the sqlParser thing), lets remove this try / catch (it has no meaning now - catching just to throw)